File: hw/palomo/u3/hw23AbstracFactory/AbstracFactory/src/main/java/ec/edu/espe/abstracfactory/view/MainView.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

package ec.edu.espe.abstracfactory.view;

import java.util.Scanner;
import ec.edu.espe.abstracfactory.model.GUIFactory;
import ec.edu.espe.abstracfactory.controller.ClientApp;

/**
 *
 * @author Cristian Palomo, Error 404, @ESPE
 */
public class MainView {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("=== Abstract Factory - GUI Multiplataforma ===");
        System.out.print("Ingrese el sistema operativo (windows/linux): ");
        String os = sc.nextLine();

        try {
            GUIFactory factory = GUIFactory.getFactory(os);

            ClientApp app = new ClientApp(factory);
            app.createInterface();
            app.renderInterface();

        } catch (IllegalArgumentException e) {
            System.out.println("Error: " + e.getMessage());
        } finally {
            sc.close();
        }
    }
}

File: hw/palomo/u3/hw23AbstracFactory/AbstracFactory.py/ec_edu_espe/view/main_view.py
from ec_edu_espe.controller.client_app import ClientApp
from ec_edu_espe.model.gui_factory import GUIFactory

def main() -> None:
    print("=== Abstract Factory - GUI Multiplataforma ===")
    os_name = input("Ingrese el sistema operativo (windows/linux): ")

    try:
        factory = GUIFactory.get_factory(os_name)

        app = ClientApp(factory)
        app.crear_interfaz()
        app.renderizar_interfaz()

    except ValueError as error:
        print(f"Error: {error}")


if __name__ == "__main__":
    main()